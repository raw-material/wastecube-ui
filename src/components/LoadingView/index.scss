.loading-view {
    padding: 0 20px;

    .button-wrapper {
        display: flex;

        button.finish-button {
            font-size: 2rem;
            position: relative;
            top: 85vh;
            display: block;
            margin: 0 auto;
        }
    }

    .pet-wrapper {
        button.pet-button {
            font-size: 3rem;
            text-align: right;
            border: 0;
            float: left;
            padding: 50px 50px;
            margin: auto;
            display: block;
        }
    }

    .hdpe-wrapper {
        button.hdpe-button {
            font-size: 3rem;
            border: 0;
            float: right;
            padding: 50px 50px;
            margin: auto;
            display: block;
        }
    }

    .vertical {
        border-left: 6px solid black;
        height: 70%;
        position: absolute;
        left: 50%;
        margin-top: 20px;
        padding-top: 20px;

    }
}
