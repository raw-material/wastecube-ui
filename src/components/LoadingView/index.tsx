import * as React from 'react';
import { useIdleTimer } from 'react-idle-timer';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faArrowLeft, faArrowRight } from '@fortawesome/free-solid-svg-icons';
import { IType } from '../App';
import './index.scss';

interface LoadingStateProps {
  changeState: (action: number) => void;
  types?: {
    left: IType;
    right: IType;
  };
}

const useIdleTimerHandler =
  process.env.NODE_ENV !== 'development'
    ? useIdleTimer
    : () => ({ reset: () => {} });

const LoadingView: React.FC<LoadingStateProps> = ({ changeState, types }) => {
  const { reset } = useIdleTimerHandler({
    timeout: 1000 * 60,
    onIdle: () => changeState(-1),
    debounce: 500
  });

  const [count, setCount] = React.useState<number>(0);

  React.useEffect(() => {
    const socket = new WebSocket('ws://localhost:5001');

    socket.addEventListener('open', (e: any) => {
      e?.data && !isNaN(e.data) && setCount(e.data);
      reset();
    });

    socket.addEventListener('message', (e: any) => {
      e?.data && !isNaN(e.data) && setCount(e.data);
    });

    return () => socket.close();
  }, []);

  return (
    <div className="loading-view">
      {types && (
        <div className="types-wrapper">
          <div className="type-column first">
            <h2>
              <FontAwesomeIcon icon={faArrowLeft} />
              <span>{types.left.title}</span>
            </h2>
            <div className="type-description">
              <figure>
                <img src={`http://localhost:5000${types.left.image}`} alt="" />
              </figure>
              <div
                dangerouslySetInnerHTML={{ __html: types.left.description }}
              />
            </div>
          </div>
          <div className="type-column second">
            <h2>
              <span>{types.right.title}</span>
              <FontAwesomeIcon icon={faArrowRight} />
            </h2>
            <div className="type-description">
              <figure>
                <img src={`http://localhost:5000${types.right.image}`} alt="" />
              </figure>
              <div
                dangerouslySetInnerHTML={{ __html: types.right.description }}
              />
            </div>
          </div>
        </div>
      )}

      <div className="count-wrapper">
        <p>Hai conferito {count} bottiglie</p>
      </div>

      <div className="button-wrapper">
        <button className="finish-button" onClick={() => changeState(1)}>
          finito
        </button>
      </div>
    </div>
  );
};

export default LoadingView;
