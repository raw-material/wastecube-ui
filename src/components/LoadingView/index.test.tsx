import React from 'react';
import * as renderer from 'react-test-renderer';
import { cleanup, render } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';
import LoadingView from './index';

expect.extend(toHaveNoViolations);
afterEach(cleanup);

it('renders loading view', () => {
  const component = renderer.create(<LoadingView changeState={() => {}} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
