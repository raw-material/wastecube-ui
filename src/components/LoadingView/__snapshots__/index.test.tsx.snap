// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loading view 1`] = `
<div
  className="loading-view"
>
  <div
    className="count-wrapper"
  >
    <p>
      Hai conferito 
      0
       bottiglie
    </p>
  </div>
  <div
    className="button-wrapper"
  >
    <button
      className="finish-button"
      onClick={[Function]}
    >
      finito
    </button>
  </div>
</div>
`;
